File: app/src/main/java/com/constraint/vagabond/retrofit/GetRecAreasImpl.java
package com.constraint.vagabond.retrofit;

import android.util.Log;

import com.constraint.vagabond.data.RecreationalAreaList;
import com.constraint.vagabond.main.search.MainContract;
import com.constraint.vagabond.network.RetrofitInstance;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class GetRecAreasImpl implements MainContract.GetRecAreasInteractor {

   private final String apiKey = "api-key";

  @Override
  public void getRecAreasList(final OnFinishedListener onFinishedListener, String query) {
    GetRecAreasData service = RetrofitInstance.getInstance().create(GetRecAreasData.class);
    Call<RecreationalAreaList> call = service.getRecreationalAreaData(query,true, apiKey);
    Log.d("URL called", call.request().url() + "");
    call.enqueue(
        new Callback<RecreationalAreaList>() {
          @Override
          public void onResponse(
              Call<RecreationalAreaList> call, Response<RecreationalAreaList> response) {
              
            onFinishedListener.onFinished(response.body());
          }

          @Override
          public void onFailure(Call<RecreationalAreaList> call, Throwable t) {
            onFinishedListener.onFailure(t);
          }
        });
  }
}

File: app/src/main/java/com/constraint/vagabond/main/search/GetRecAreasImpl.kt
package com.constraint.vagabond.main.search

import android.util.Log

import com.constraint.vagabond.data.entities.RecreationalAreaList
import com.constraint.vagabond.network.RetrofitInstance

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class GetRecAreasImpl : MainContract.GetRecAreasInteractor {

    override fun getRecAreasList(onFinishedListener: MainContract.GetRecAreasInteractor.OnFinishedListener, query: String) {
        val service = RetrofitInstance.create()
        val apiKey = "api-key"
        val call = service.getRecreationalAreaData(query, true, apiKey)
        Log.d("URL called", call.request().url().toString() + "")
        call.enqueue(
                object : Callback<RecreationalAreaList> {
                    override fun onResponse(call: Call<RecreationalAreaList>, response: Response<RecreationalAreaList>) {
                        onFinishedListener.onFinished(response.body())
                    }

                    override fun onFailure(call: Call<RecreationalAreaList>, t: Throwable) {
                        onFinishedListener.onFailure(t)
                    }
                })
    }
}
